### DISCUSSION TITLE
Please add “Visual Studio Code - Insiders” to Editor preference

### DISCUSSION THREAD

**Discussion by dtinth** [community/community#5163] [Codespaces]
When **Editor preference** is set to **Visual Studio Code** but I only have **Visual Studio Code - Insiders** installed…

![image](https://user-images.githubusercontent.com/193136/129870071-5e503ab2-842f-4e09-a854-7c83fde10af8.png)

When I click on **New codespace** button, after the button changes to loading state, nothing happens. I try to click on it again. Still nothing happens.

![recording](https://user-images.githubusercontent.com/193136/129870701-2ef2f804-ea30-4e23-85c0-17b9e67335fc.gif)

When I refresh, it turns out that the Codespace is created twice.

![image](https://user-images.githubusercontent.com/193136/129870731-f51b61f3-4e0a-4939-9242-648ece2209d4.png)

Is there any way to make this work with “Code - Insiders”?

**Comment by ghost** [comment]
I want to be kept posted on this discussion as well. Gentlemen, please add the VSCode insiders as a choice.

**Comment by Dmutro777** [comment]
Вы получили письмо от пользователя Dmutro Bezdetniy в конфиденциальном режиме Gmail:

[image: Логотип Gmail] <https://confidential-mail.google.com/msg/AGMme3Oqj2i3kdzruLTleXhXoQjU-uhqQN9T6ldDWkwdnKIhoEuBJ0P64GUb8Y9YGWibfc1drsE4ogir1sZb8O257f1lnx2b1XWs6hbRRhJn_R7uxBLUo3cUCHxIE9vTYdNZqy7-CHd70lrnldPUY9piLpnX5C7rmy6UwyOm6eDLWdJVpvA89GdD7pjlVkRXrsbpjtEFTT0feQ1cYto=>

Письмо отправлено 23 окт. 2022 г. в 17:39:05 GMT-7
Чтобы его открыть, перейдите по ссылке ниже. Доступно только в аккаунте ***@***.***

Открыть письмо <https://confidential-mail.google.com/msg/AGMme3Oqj2i3kdzruLTleXhXoQjU-uhqQN9T6ldDWkwdnKIhoEuBJ0P64GUb8Y9YGWibfc1drsE4ogir1sZb8O257f1lnx2b1XWs6hbRRhJn_R7uxBLUo3cUCHxIE9vTYdNZqy7-CHd70lrnldPUY9piLpnX5C7rmy6UwyOm6eDLWdJVpvA89GdD7pjlVkRXrsbpjtEFTT0feQ1cYto=>

Конфиденциальный режим Gmail позволяет настроить параметры безопасности отправляемого письма. Отправитель может задать срок доступа к письму, запретить печать или пересылку и отслеживать доступ к письму. Подробнее… <https://support.google.com/mail/answer/7674059>
Gmail – это почтовый сервис от Google
Применяется Политика конфиденциальности Google <https://myaccount.google.com/privacypolicy?hl=ru>
Google LLC, 1600 Amphitheatre Parkway, Mountain View, CA 94043, USA
Вы получили это сообщение, так как Вам отправили письмо в конфиденциальном режиме Gmail.
[image: Логотип Google]

**Comment by jdhenry08** [comment]
Well, this isn't promising. Came looking for an answer to whether this was possible. No update in 3 years; seems unlikely that it's a priority.

**Comment by billmcilhargey** [comment]
Please add the Visual Studio Code Insider as a default option +1

Recommended workaround for now:

**Set GitHub Codespaces Editor to “Visual Studio Code for the Web”**

1. Go to GitHub Settings Visit https://github.com/settings/codespaces
2. Scroll to “Editor preference” and you’ll see three options:

- Visual Studio Code (desktop)
- Visual Studio Code for the Web
- JupyterLab (Preview)

3. Select “Visual Studio Code for the Web” - This ensures that when you launch a Codespace, it opens in the browser-based VS Code editor.

<img width="775" height="220" alt="Screenshot 2025-08-13 at 7 41 16 AM" src="https://github.com/user-attachments/assets/abe07f46-451a-4ddf-9163-e36d173bc2ba" />



**Use VS Code Web → Reopen in Desktop (Insiders)** - (this is GitHub’s officially supported path)

1. From within your GitHub Repro, open your Codespace in VS Code Web (click on the name of the codespace if its the default), otherwise, choose “Visual Studio Code for the Web” when launching a Codespace.

2. In the browser editor, click “Reopen in Desktop” (top right left under the three lines menu - sometimes referred to as the hamburger menu).

3. When prompted, choose Open in VS Code Insiders Desktop.

<img width="334" height="343" alt="Screenshot 2025-08-13 at 7 46 35 AM" src="https://github.com/user-attachments/assets/0dda6eca-39fb-4c7b-ae0e-f736b6f9d049" />

✅ This method works consistently and doesn’t require protocol hacking.
